pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -573,91 +587,96 @@ def newpkgcb(pkgId, name, arch):
                         packages_pb.state = 'running'
                         packages_pb.save()
 
-                        for pkg in packages.values():
-                            package = Package(**Package.createrepo_to_dict(pkg))
-                            artifact = Artifact(size=package.size_package)
-                            checksum_type = getattr(CHECKSUM_TYPES, package.checksum_type.upper())
-                            setattr(artifact, checksum_type, package.pkgId)
-                            url = urljoin(remote_url, package.location_href)
-                            filename = os.path.basename(package.location_href)
-                            da = DeclarativeArtifact(
-                                artifact=artifact,
-                                url=url,
-                                relative_path=filename,
-                                remote=self.remote,
-                                deferred_download=self.deferred_download
-                            )
-                            dc = DeclarativeContent(content=package, d_artifacts=[da])
-                            dc.extra_data = defaultdict(list)
-
-                            # find if a package relates to a modulemd
-                            if dc.content.nevra in nevra_to_module.keys():
-                                dc.content.is_modular = True
-                                for dc_modulemd in nevra_to_module[dc.content.nevra]:
-                                    dc.extra_data['modulemd_relation'].append(dc_modulemd)
-                                    dc_modulemd.extra_data['package_relation'].append(dc)
-
-                            if dc.content.name in pkgname_to_groups.keys():
-                                for dc_group in pkgname_to_groups[dc.content.name]:
-                                    dc.extra_data['group_relations'].append(dc_group)
-                                    dc_group.extra_data['related_packages'].append(dc)
-
-                            packages_pb.increment()
-                            await self.put(dc)
+                        for batch in grouper(list(packages.values()), 250):
+                            packages_pb.increase_by(len(batch))
+
+                            for pkg in batch:
```

Comment:
Hard to tell from the diff but the only change is these 3 lines and the removal of .increment() below

Reply:
I took a time measurement of the total time spent in this loop before changing anything, and measured that it took about 30 seconds with 10,500 packages.  So 250 packages should represent about 1 update per second, roughly (on my machine).